hs_core/tasks.py

```diff
@@ -367,6 +369,16 @@ def create_temp_zip(resource_id, input_path, output_path, aggregation_name=None,
             head, tail = os.path.split(temp_folder_name)  # tail is unqualified folder name "foo"
             out_with_folder = os.path.join(temp_folder_name, tail)  # foo/foo is subdir to zip
             istorage.copyFiles(input_path, out_with_folder)
+            if not aggregation:
```

Comment:
I'll look into this task and see if I can do some of the general refactoring I'm talking about.  I'm more familiar with this piece of code and it'd be a good study for me.